src/raftstore/store/store.rs

```diff
@@ -500,12 +501,37 @@ impl<T: Transport, C: PdClient> Store<T, C> {
             has_peer = false;
         }
 
-        if !has_peer {
-            let peer = try!(Peer::replicate(self, region_id, target.get_id()));
-            // We don't have start_key of the region, so there is no need to insert into
-            // region_ranges
-            self.region_peers.insert(region_id, peer);
+        if has_peer {
+            return Ok(true);
         }
+
+        let message = msg.get_message();
+        let msg_type = message.get_msg_type();
+        if msg_type != MessageType::MsgRequestVote &&
+           (msg_type != MessageType::MsgHeartbeat || message.get_commit() != INVALID_INDEX) {
+            warn!("target peer {:?} doesn't exist, stale message {:?}.",
+                  target,
+                  msg_type);
+            return Ok(false);
+        }
+
+        let start_key = data_key(msg.get_start_key());
```

Comment:
That's wired, I suggest we set the start key of first region to 0x0, and set the end key to 0xFF.
I just reviewed the currently code, our tests rely on empty start key and end key.